sdk/resource/resource.go

```diff
@@ -192,6 +249,31 @@ func Merge(a, b *Resource) (*Resource, error) {
 	return merged, nil
 }
 
+// MergeAt creates a new resource by combining resources at the target
+// schemaURL version.
+//
+// If there are common keys between resources the latter resource will
+// overwrite the former.
+//
+// Any of the resources not already at schemaURL version will be appropriately
+// upgraded or downgraded to match the version. An error is returned if this is
+// not possible.
```

Comment:
Document as a warning that users need to be sure about _all_ the schema URLs this will reach out to.